### PR TITLE
Refactor makefiles and use whalephant

### PR DESCRIPTION
**Pull request by Niktux**
# Features
- Refactor makefiles
- Use whalephant for phpunit dockerfile
- Use .env to handle WEB_PORT nicely
- Define real "clean" target
- Fix gitignore composer.lock issue
- Fix ``make connect`` (no more hardcoded container name)

# Refactor makefiles
- Gather all included makefiles into makefiles directory
- Each tool has its own makefile
- Same skeleton for each files : header / declarations / targets / clean targets / phony target (separated by comment lines)
- Try to make each file as independant as possible (that's why cli_args preprocessing is duplicated)
- Self-documented makefiles (defautl target is now the help one)

# Stuffs
- To display make help : ``make`` or ``make help``
- To test : ``make init up``
- To re-test : ``make clean init up``
- To test whalephant : ``make phpunit``
- To change WEB_PORT after ``make init``: ``vi .env``

# Advices
Diff may be quite hard to read. Look at branch to see the expected result

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+WEB_PORT=80
+COMPOSE_PROJECT_NAME=onyx
```
Comment (lebris) [inline .env.example]: Wizard à mettre à jour. https://github.com/Onyx-org/core/blob/master/wizards.mk#L27

Comment (Niktux) [inline .env.example]: fixed in 0.1.1

File: makefiles/whalephant.mk [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#------------------------------------------------------------------------------
+# Whalephant
+#------------------------------------------------------------------------------
+
+whalephant = docker run -it --rm --name whalephant \
+                             -v ${HOST_SOURCE_PATH}:${CONTAINER_SOURCE_PATH} \
```
Comment (lebris) [inline makefiles/whalephant.mk]: `CONTAINER_SOURCE_PATH` est défini dans `phpunit.mk`

Mais ici `CONTAINER_SOURCE_PATH` est indépendant pour whalephant, il pourrait être en dur dans ce makefile.

Comment (Niktux) [inline makefiles/whalephant.mk]: Agree

Comment (lebris) [inline makefiles/whalephant.mk]: Fix 4932fa307311e78c61773df8f89a916cb9c2db59

**Pull request merged**